tasks/managed/filter-already-released-images/filter-already-released-images.yaml

```diff
@@ -0,0 +1,344 @@
+          - name: pathInRepo
+            value: stepactions/use-trusted-artifact/use-trusted-artifact.yaml
+      params:
+        - name: workDir
+          value: $(params.dataDir)
+        - name: sourceDataArtifact
+          value: $(params.sourceDataArtifact)
+    - name: filter-already-released-images
+      image: quay.io/konflux-ci/release-service-utils:02a8ddcd16113371a255fd1ef0a196399d300162
+      computeResources:
+        limits:
+          memory: 1Gi
+        requests:
+          memory: 1Gi
+          cpu: 250m
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        SNAPSHOT_FILE="$(params.dataDir)/$(params.snapshotPath)"
+        DATA_FILE="$(params.dataDir)/$(params.dataPath)"
+
+        # Helper to detect "not found" style errors from oras that should be treated as
+        # an absent tag instead of a fatal registry error.
+        is_not_found_error() {
+            local err_msg="${1:-}"
+            if grep -qiE "manifest unknown|name unknown|was not found|not found" <<< "${err_msg}"; then
+                return 0
+            fi
+            return 1
+        }
+
+        if [ ! -f "${SNAPSHOT_FILE}" ]; then
+            echo "Error: Snapshot file not found: ${SNAPSHOT_FILE}"
+            exit 1
+        fi
+
+        if [ ! -f "${DATA_FILE}" ]; then
+            echo "Error: Data file not found: ${DATA_FILE}"
+            exit 1
+        fi
+
+        SNAPSHOT_JSON=$(cat "${SNAPSHOT_FILE}")
+        COMPONENT_COUNT=$(jq '.components | length' <<< "${SNAPSHOT_JSON}")
+
+        FILTERED_COMPONENTS='[]'
+        FILTERED_COUNT=0
+
+        for ((i=0; i<COMPONENT_COUNT; i++)); do
+            COMPONENT=$(jq -c ".components[$i]" <<< "${SNAPSHOT_JSON}")
+            COMPONENT_NAME=$(jq -r '.name' <<< "${COMPONENT}")
+            CONTAINER_IMAGE=$(jq -r '.containerImage' <<< "${COMPONENT}")
+
+            # Get the source image digest using oras resolve (same as push-snapshot)
+            # This ensures we compare manifest index digests, not platform-specific ones
+            SOURCE_AUTH_FILE=$(mktemp)
+            if ! select-oci-auth "${CONTAINER_IMAGE}" > "${SOURCE_AUTH_FILE}" 2>/dev/null || \
+               [ ! -s "${SOURCE_AUTH_FILE}" ]; then
+                echo '{}' > "${SOURCE_AUTH_FILE}"
+            fi
+
+            SOURCE_ERROR_FILE=$(mktemp)
+            if ! DIGEST=$(oras resolve --registry-config "${SOURCE_AUTH_FILE}" \
+                "${CONTAINER_IMAGE}" 2>"${SOURCE_ERROR_FILE}"); then
+                SOURCE_ERROR_MSG=$(cat "${SOURCE_ERROR_FILE}")
+                rm -f "${SOURCE_ERROR_FILE}" "${SOURCE_AUTH_FILE}"
+                
+                # Check if it's a benign error (image not found in registry)
+                if is_not_found_error "${SOURCE_ERROR_MSG}"; then
+                    # Treat as "not yet released", keep component for release
+                    echo "WARNING: Source image ${CONTAINER_IMAGE} not found in registry, treating as not yet released"
```

Comment:
Can we not call this a `source image`? It is confusing with the source container (like the .src one)

Reply:
I am fine with that, as long as we don't say `source`. Thanks